it/pages/changePref/EmailToUsePageSpec.scala

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages.changePref
+
+import assets.BaseITConstants.internalServerErrorTitle
+import common.SessionKeys
+import forms.YesNoForm
+import helpers.SessionCookieCrumbler
+import models.{No, Yes, YesNo}
+import pages.BasePageISpec
+import play.api.libs.ws.WSResponse
+import play.api.http.Status
+import stubs.VatSubscriptionStub
+
+class EmailToUsePageSpec extends BasePageISpec {
+
+  val path = "/preference-confirm-email"
+
+  "Calling the EmailToUse .show method" when {
+
+    def show: WSResponse = get(path)
+
+    "the user is authenticated" when {
+
+      "a success response is received for Customer Details" when {
+
+        "the data is valid" should {
+
+          "display the page correctly" in {
+
+            given.user.isAuthenticated
+
+            And("a successful response for an individual is stubbed")
+            VatSubscriptionStub.stubCustomerInfo
+
+            When("the emailToUse page is called")
+            val result = show
+
+            result should have(
+              httpStatus(Status.OK),
+              pageTitle(generateDocumentTitle("emailToUse.title"))
+            )
+          }
+
+          "add the email to session" in {
+
+            given.user.isAuthenticated
+
+            And("a successful response for an individual is stubbed")
+            VatSubscriptionStub.stubCustomerInfo
+
+            When("a valid email is submitted")
+            val res = show
+
+            SessionCookieCrumbler.getSessionMap(res).get(SessionKeys.prepopulationEmailKey) shouldBe Some("testemail@test.com")
+            SessionCookieCrumbler.getSessionMap(res).get(SessionKeys.validationEmailKey) shouldBe Some("testemail@test.com")
+          }
+        }
+
+        "the data is invalid" should {
+
+          "return an error if invalid data is received" in {
+
+            given.user.isAuthenticated
+
+            And("a success response for Customer Details with invalid data is stubbed")
+            VatSubscriptionStub.stubCustomerInfoInvalidJson
+
+            When("the emailToUse page is called")
+            val result = show
+
+            result should have(
+              httpStatus(Status.INTERNAL_SERVER_ERROR),
+              pageTitle(generateDocumentTitle(internalServerErrorTitle))
+            )
+          }
+        }
+
+
+      }
+
+      "an error response is received for Customer Details" in {
```

Comment:
I'd argue this and the invalid data test before it are so similar that you only need one. Looking at the whole integration flow, the only difference between those two scenarios is in the HTTP parser. If all the other steps in the components are the same then this seems like it's already covered by a unit test for the HTTP parser. What you think?